pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -350,7 +351,12 @@ public Representation post(Representation entity) {
 
           // 3/ Request is parsed by the handler which generates a
           // list of FileItems
-          items = upload.parseRequest(getRequest());
+          try {
+            items = upload.parseRequest(getRequest());
+          } catch (FileUploadBase.IOFileUploadException e) {
+            LOGGER.info("Exception uploading segment: " + e);
```

Comment:
parseRequest doc says it throws FileUploadException which can indicate bad request format, early connection termination or local disk issue.  
For bad request format  it throws FileUploadBase.InvalidContentType exception that should be info logged with correct return status and message.
IOFileUploadException indicates issue reading network request or writing to local disk. That should be error. Even for connection termination it's hard to say if that's client issue.